src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -59,20 +61,49 @@ public Task<bool> NominateProjectAsync(string projectUniqueName, CancellationTok
         }
 
         public Task<bool> NominateProjectAsync(string projectUniqueName, IVsProjectRestoreInfo projectRestoreInfo, CancellationToken token)
+        {
+            return NominateProjectAsync(projectUniqueName, projectRestoreInfo, null, token);
+        }
+
+        public Task<bool> NominateProjectAsync(string projectUniqueName, IVsProjectRestoreInfo2 projectRestoreInfo, CancellationToken token)
+        {
+            return NominateProjectAsync(projectUniqueName, null, projectRestoreInfo, token);
+        }
+
+        /// <summary>
+        /// This is where the nominate calls for the IVs1 and IVS3 APIs combine. The reason for this method is to avoid duplication and potential issues 
+        /// The issue with this method is that it has some weird custom logging to ensure backward compatibility. It's on the implementer to ensure these calls are correct.
```

Comment:
Do we have tests that cover the "weird custom logging to ensure backward compatibility"?  My concern is that if someone other than you (like me) is working in this code, they might not understand the requirements.  However, if the requirements are enforced through unit tests, the requirements should be clear from well-named tests and expected results.

Reply:
So I looked, it doesn't cover all scenarios. 

I will go over them again and make sure they are covered.